pkg/snapshotcombiner/snapshotcombiner.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 The Inspektor Gadget authors
```

Comment:
Does this come from https://github.com/inspektor-gadget/inspektor-gadget/pull/898? I suggest merging that one first and basing this PR on it.

Reply:
Yes, it's based on that but got some updates already and is a necessary part (even if we don't use the caching, which was still in discussion) of this PR. I'm fine with updating that and decoupling, if that helps.

Reply:
We agreed to keep in in here, so I'm resolving this.